rediscluster/mapping.go

```diff
@@ -57,6 +73,33 @@ func (c *Cluster) newNode(addr string, initial bool) (*node, error) {
 	return node, nil
 }
 
+func (c *Cluster) nodeOpts(addr string) (*redisconn.Opts, error) {
+	nodeOpts := c.opts.HostOpts
+
+	if !nodeOpts.TLSEnabled {
+		return &nodeOpts, nil
+	}
+
+	originalHost, err := redisclusterutil.GetHost(addr)
+	if err != nil {
+		return nil, err
+	}
+
+	if !redisclusterutil.IsIPAddress(originalHost) {
+		// preserve original hostname for TLS verification
+		if nodeOpts.TLSConfig != nil {
+			nodeOpts.TLSConfig = nodeOpts.TLSConfig.Clone()
+			nodeOpts.TLSConfig.ServerName = originalHost
```

Comment:
This is a common code between the two branches, but it doesn't seem that way. I think we should move it outside of this `if`.

Reply:
fixed